akka-actor-typed-tests/src/test/scala/akka/actor/typed/scaladsl/ActorContextAskSpec.scala

```diff
@@ -156,6 +156,37 @@ class ActorContextAskSpec
       exc.getMessage should include("[10 ms]") // timeout
     }
 
+    "receive replies in same order as sent" in {
```

Comment:
This test failed (sometimes) without the fix.
Also tried it with higher N values.